src/Modis09L2GeoFile.cpp

```diff
@@ -118,24 +121,28 @@ Modis09L2GeoFile::readFile(const std::string fileName, int verbose, int quiet,
 						   sizeof(unsigned long long))))
 	return SSC_ENOMEM;
 
-    int level = 5;
+    int level = 27;
     STARE index(level, build_level);
```

Comment:
I believe, with your changes, that this line can be removed, which should improve performance...